### PR TITLE
Feature/gh 8704 table layout

### PR DESCRIPTION
**Pull request by tobi-or-not-tobi**
Improved table configurations: 
- global component configurations
- breakpoint configuration merging (small to big)
- allow for default config structure in `TableService.getTableConfig`
- various table layouts (vertical, vertical-stacked, horizontal)

### CODE REVIEW

File: feature-libs/my-account/organization/src/components/unit/addresses/details/current-unit-address.service.ts [hunk 1]
```diff
@@ -8,8 +8,8 @@ import {
   switchMap,
   withLatestFrom,
 } from 'rxjs/operators';
-import { OrgUnitService } from '../../../../core/services/org-unit.service';
 import { CurrentUnitService } from '../../current-unit.service';
+import { OrgUnitService } from '../../../../core/services/org-unit.service';
```
Comment (dunqan) [inline feature-libs/my-account/organization/src/components/unit/addresses/details/current-unit-address.service.ts]: Deep import

File: projects/storefrontlib/src/shared/components/table/table-data-cell/table-data-cell.component.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ import { TableHeaderOutletContext } from '../table.model';
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
 export class TableDataCellComponent {
+  @HostBinding('class') cls = 'content-wrapper';
```
Comment (dunqan) [inline projects/storefrontlib/src/shared/components/table/table-data-cell/table-data-cell.component.ts:10]: Is this required for all table cells components?

File: feature-libs/my-account/organization/src/components/shared/organization-list/organization-list.service.ts [hunk 3]
```diff
@@ -87,7 +102,10 @@ export abstract class OrganizationListService<T, P = PaginationModel> {
    */
   protected getStructure(): Observable<TableStructure> {
     return combineLatest([
-      this.tableService.buildStructure(this.tableType),
+      this.tableService.buildStructure(
+        this.tableType,
+        this.defaultTableStructure
```
Comment (dunqan) [inline feature-libs/my-account/organization/src/components/shared/organization-list/organization-list.service.ts:107]: We are not doing deepMerge here, is it OK?

File: projects/storefrontlib/src/shared/components/table/table.service.ts [hunk 4]
```diff
@@ -119,41 +93,87 @@ export class TableService {
...
 
   /**
-   * Prints a convenient message in the console to increase developer experience.
+   * Logs a message in the console to increase developer experience.
+   *
+   * The message is only logged in dev mode.
    */
-  private warn(message) {
+  private logWarning(message) {
     if (isDevMode) {
```
Comment (dunqan) [inline projects/storefrontlib/src/shared/components/table/table.service.ts]: `isDevMode()`?

File: projects/storefrontlib/src/shared/components/table/table-data-cell/table-data-cell.component.ts (hunk 2 above)
Comment (dunqan) [reply projects/storefrontlib/src/shared/components/table/table-data-cell/table-data-cell.component.ts:10]: ok, nevermind ;-)

File: feature-libs/my-account/organization/src/components/shared/organization-list/organization-list.service.ts (hunk 3 above)
Comment (tobi-or-not-tobi) [reply feature-libs/my-account/organization/src/components/shared/organization-list/organization-list.service.ts:107]: It's ok for now. Might be improved over time

**Pull request merged**